package/src/components/ChannelList/hooks/listeners/useNewMessage.ts

```diff
@@ -24,12 +26,26 @@ export const useNewMessage = <
   useEffect(() => {
     const handleEvent = (event: Event<StreamChatGenerics>) => {
       setChannels((channels) => {
-        if (!lockChannelOrder && event.cid) {
-          return moveChannelUp<StreamChatGenerics>({
-            channels,
-            cid: event.cid,
-          });
+        if (!lockChannelOrder && event.cid && event.channel_type && event.channel_id) {
+          const targetChannelIndex = channels.findIndex((c) => c.cid === event.cid);
+
+          if (targetChannelIndex >= 0) {
+            return moveChannelUp<StreamChatGenerics>({
+              channels,
+              cid: event.cid,
+            });
+          }
+
+          // If channel doesn't exist in existing list, check in activeChannels as well.
+          // It may happen that channel was hidden using channel.hide(). In that case
+          // We remove it from `channels` state, but its still being watched and exists in client.activeChannels.
+          const channel = client.channel(event.channel_type, event.channel_id);
+
+          if (channel.initialized) {
+            return uniqBy([channel, ...channels], 'cid');
```

Comment:
`uniqBy([channel, ...channels], 'cid');`

the array doesnt have the hidden channel since the targetIndex was < 0, so `[channel, ...channels]` should suffice?

Reply:
You are right, pardon me for shameless copy&paste!!